sympy/integrals/integrals.py

```diff
@@ -1596,3 +1603,61 @@ def line_integrate(field, curve, vars):
 
     integral = Integral(Ft, curve.limits).doit(deep=False)
     return integral
+
+
+def pole_reintegrate(f, x, antideriv,
+                     reintegrator=lambda f, x: integrate(f, x, conds="none")):
```

Comment:
Another concern is that if someone passes a specific algorithm to integrate, like integrate(meijerg=True), should the reintegrator also only use that algorithm? I'm unsure. It does seem to make sense for the flag to be passed through, but it could lead to the confusing situation where `integrate(f, x)` is computed with, say, meijerg, but `integrate(f, x, meijerg=True)` doesn't produce the same result, because meijerg isn't what is actually used by the default reintegrator.